### ISSUE TITLE
Specifying a resource group with a - in the name fails on deployment template validation

### ISSUE THREAD

**Issue by chadbert** [Azure-Samples/Cognitive-Services-Voice-Assistant#437]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>
1. Clone repository
1. cd to scripts directory
1. run az login
1. run az account set -s <subscription name>
1. deploy using ./deployAll.ps1 -appName hospitality -resourceName speech-demo -region WestUS

### Any log messages given by the failure
> Yes. It is nowhere near apparent enough to tell you what the issue is.
Please enter 'y' to proceed, or any other character to quit:
y
Calling deployTemplate
.\deployTemplate.ps1 -resourceName speech-demo -region WestUS -luisName speech-demo-6159 -functionName speech-demo-6159 -storageName speech-demo6159
Creating resource group
Deploying azure template at ./azuredeploy.json
{'additional_properties': {}, 'code': 'InvalidTemplateDeployment', 'message': "The template deployment 'azuredeploy' is not valid according to the validation procedure. The tracking id is 'e264770c-1ea2-4515-8016-17023db1aebb'. See inner errors for details.", 'target': None, 'details': [<azure.mgmt.resource.resources.v2019_07_01.models._models.ErrorResponse object at 0x04F95150>], 'additional_info': None}
Write-Error: Failed to deploy template

### Expected/desired behavior
> Expected behavior is to either not allow a - in the name or to strip out the - when creating the storage account. I only figured out what the issue was by manually installing the deployment template and executing it within the portal itself. What is occurring is that the storage account validation fails because it doesn't allow the '-' character. In the temp parameters file, it is setting this as the parameter for the storage account:
"storageName": {
  "value": "speech-demo6159"
}

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)
Windows 10

### Versions
>
Windows 10 1909
source: latest from master as of 5/21/2020

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by dargilco** [comment]
Hi @chadbert, thanks for your feedback! We do mention in the [README.md](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/custom-commands/README.md) "This name should be no more than 19 characters, alphanumeric only" but you are right that the script should do a better job enforcing this. @chschrae may also want to chime in.
Darren

**Comment by chschrae** [comment]
Yup 100% agree. We are working to improve the script's robustness and will be sure to add this.
